### PR TITLE
Added e-mail registration

### PR DESCRIPTION
**Pull request by deleterium**
Hi Mark!

Here's a contribution to add an option to use e-mail registration. It was added a new table to the database (unverified_users) and, when this option is enabled (in mail_config.php), the registration is done to this table. Once the user check his e-mail, he will use link to verify (verify.php) the registration and his information will be inserted to the table of active users (function verifyHelper). This time I also added information to the documentation :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ A webserver with php and an SQL server is needed to run PostQ.
 3. Rename `sqlconfig.php_example` to `sqlconfig.php`.
 4. Open `/install.php` in browser. This will create the necessary database tables.
 5. Open `index.html` and use the application.
+6. E-mail verification can be enabled editing file `mail_config.php`. Remeber to configure PHP mail() function to be able to send e-maisl from your server.
```
Comment (markszabo) [inline README.md]: Typo in the end: `e-maisl`

File: mail_config.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+$expiration_time = 60*60; //time in seconds
+
+$mail_subject = 'Verification code'; // Give the email a subject 
+$mail_message = '
+ 
+Thanks for signing up!
+Your account has been created, you can login after activating it using url below.
+ 
+Please click this link to activate your account:
+http://www.yourwebsite.com/verify.php?code=<registration_code/>
```
Comment (markszabo) [inline mail_config.php]: Very minor thing, but using `https` might stop people leaving it as `http` and then potentially leaking the URL.

File: register.php [hunk 3]
```diff
@@ -19,15 +19,52 @@
...
+include_once("mail_config.php");
+
 if(!$_POST['username'] || !$_POST['password'] || !$_POST['privatekey'] || !$_POST['publickey'])
   die("Error - one of the parameters is not set.");
 
 //check if user already exists
 if(userExists($conn, $_POST['username']))
   die("Error - user already exists. Try to login instead.");
 
+if ($use_mail_registration) {
```
Comment (markszabo) [inline register.php]: It's really just nitpicking, but the rest of the code doesn't seem to have a space after if, so keeping that convention would be ideal.

**Review by markszabo — COMMENTED** [review]
Thank you again for the great contribution. I left some minor comments, but overall it looks good. I would have probably done a little differently, but your solution is also fine.

In case you are interested, I would probably have created a `verify_email` table with the following fields: `userid`, `registrationcode`, `expiration`, and then add the users to the `users` table on registration anyway, but also check in `verify_email` on login whether they are verified or not. This would keep the database structure and the code a little bit cleaner, but overall your solution also gets the job done and and enables easier cleanup if the number of fake registration without email verification is significant. So you can keep it your way.

**Comment by deleterium** [comment]
Agreed to all comments! Files updated.

**Review by markszabo — APPROVED** [review]
Great, thanks.

**Comment by markszabo** [comment]
Hi,

I just added a license to this project, let me know if you are fine with it: [https://github.com/markszabo/PostQ/pull/5](https://github.com/markszabo/PostQ/pull/5)

**Pull request merged**